agent/ecscni/netconfig_linux.go

```diff
@@ -192,12 +195,37 @@ func NewServiceConnectNetworkConfig(
 	var egressConfig *EgressConfigJSON
 	if scConfig.EgressConfig != nil {
 		egressConfig = &EgressConfigJSON{
-			ListenerPort: scConfig.EgressConfig.ListenerPort,
+			RedirectMode: string(redirectMode),
 			VIP: VIPConfigJSON{
 				IPv4CIDR: scConfig.EgressConfig.VIP.IPV4CIDR,
 				IPv6CIDR: scConfig.EgressConfig.VIP.IPV6CIDR,
 			},
 		}
+		switch redirectMode {
+		case NAT:
+			// NAT redirect mode is for awsvpc tasks, where the one and only pause container netns will have a NAT redirect rule.
+			egressConfig.ListenerPort = scConfig.EgressConfig.ListenerPort
+		case TPROXY: // bridge
+			// TPROXY redirect mode is used for bridge-mode tasks. There are two use cases:
+			// 1. SC pause container netns will set up TPROXY that requires the Egress port
+			// 2. Other task pause container netns will add a route for traffic destined for SC VIP-CIDR to go to SC container.
+			//    In that case the configuration requires the SC (pause) container IP.
+			if shouldIncludeRedirectIP {
+				scPauseIPConfig := scConfig.RuntimeConfig.PauseContainerIPConfig
```

Comment:
That's a good callout - no it won't be null if we already get here, but it might be null when we populate pause container IPs so I'll add a check there.

Reply:
Created a new field `NetworkConfig` that's no longer nested in `RuntimeConfig` to avoid the complication that multiple containers could potentially be creating it.